### PR TITLE
Refactor our Team Compass and update onboarding documentation

### PR DESCRIPTION
**Pull request by choldgraf**
This PR is a light refactoring of our Team Compass information with the following goals:

- Organize it so that there's a natural top-level section for everything, and information is nested under the right top-level area
- Remove outdated information
- Remove redundant information and instead cross-link to sources of truth
- Update our onboarding documentation following these principles
- Try to cut down on wordiness in general

It also changes our documentation linkchecker to use the Lychee action instead of `make linkcheck`, which lets us have nice markdown linkchecker summaries in our `Job Summary` tab.

preview here: https://2i2c-team-compass--449.org.readthedocs.build/en/449/

closes https://github.com/2i2c-org/team-compass/issues/460

ref: https://github.com/2i2c-org/meta/issues/337

### CODE REVIEW

**Comment by choldgraf** [comment]
cc @jmunroe - it would be helpful to know if there's anything obvious missing here, or that could be clearer, given that you've recently experienced all of this for the first time!

File: about/governance.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  This is generally done via opening GitHub issues.
+- Team members should have time to understand the proposal, and ask questions about it to understand its ramifications.
+- Team members should have time to object and make suggested changes.
+- The result of the decision should be recorded somewhere that is available to all team members.
+
+Generally speaking, this process is carried out via GitHub Issues and Pull Requests.
+See [](development:merge-policy) for how this works in practice.
+
+[^consent-decision-making]:
+
```
Comment (jmunroe) [inline about/governance.md:21]: Formatting error on the `[^consent-decision-making]` footnote. Remove newline?

**Review by jmunroe — CHANGES_REQUESTED** [review]
Overall this is a much better structure to the documents.  With respect to the details, I'll revisit this material again after trying to applying the recommended organizational practices but that shouldn't stop the PR from being merged.

Just one smalling formating issue as noted above to fix.

**Comment by choldgraf** [comment]
Thanks for the feedback @jmunroe - I've fixed the problem that you noted!

Since this PR is a re-structure, and doesn't change a ton of content so much as re-organizes it, I am going to merge this one and we can iterate in follow-up PRs since this will get out of date quickly. Maybe @jmunroe and I can have some docs brainstorms as we work through https://github.com/2i2c-org/meta/issues/337

**Pull request merged**